src/components/common/delete-widget/delete-widget.directive.coffee

```diff
@@ -0,0 +1,20 @@
+module = angular.module('impac.components.common.delete-widget',[])
+
+module.directive('commonDeleteWidget', ($templateCache, ImpacWidgetsSvc, ImpacUtilities) ->
+  return {
+    restrict: 'A',
+    template: $templateCache.get('common/delete-widget.tmpl.html'),
+    scope: {
+      parentWidget: '='
+      onDismiss: '&'
+    }
+    link: (scope, element) ->
+      scope.loading = false
+
+      scope.deleteWidget = ->
+        scope.loading = true
+        ImpacWidgetsSvc.delete(scope.parentWidget)
```

Comment:
It should be the parents (`impacWidget`) responsibility to make the backend request to delete the widget. This component should just emit an event via an `onDelete` callback.